lib/grp.gi

```diff
@@ -1586,6 +1586,45 @@ function( G )
     return comp;
 end );
 
+#############################################################################
+##
+#M  Socle( <G> )  . . . . . . . . . . . . . . . . for finite nilpotent groups
+##
+InstallMethod( Socle, "for finite nilpotent groups",
+              [ IsGroup ],
+              RankFilter( IsGroup and IsFinite and IsNilpotentGroup ) -
+              RankFilter( IsGroup ),
+  function(G)
+    local H, prodH;
+
+    if not IsFinite(G) or not IsNilpotentGroup(G) then
+      TryNextMethod();
+    fi;
```

Comment:
Won't this usually end up computing the size of the group? That in turn may not terminate. Perhaps have a `CanComputeSize` check first, perhaps as part of the filter?

An easy way to trigger a problem:
```
F:=FreeGroup(2);; G:=F/[F.1^2,F.2^2]; Socle(G);
```